xbmc/pvr/dialogs/GUIDialogPVRTimerSettings.cpp

```diff
@@ -759,8 +759,11 @@ void CGUIDialogPVRTimerSettings::InitializeTypesList()
     if (type->ForbidsEpgTagOnCreate() && m_timerInfoTag->GetEpgInfoTag())
       continue;
 
-    // Drop TimerTypes that aren't rules if end time is in the past
-    if (!type->IsTimerRule() && m_timerInfoTag->EndAsLocalTime() < CDateTime::GetCurrentDateTime())
+    EPG::CEpgInfoTagPtr epgTag = m_timerInfoTag->GetEpgInfoTag(false);
+    //If we have an epg tag, ask the pvr addon if recording is possible, else, use the current time
+    bool canRecord = epgTag ? epgTag->IsRecordable() : m_timerInfoTag->EndAsLocalTime() >= CDateTime::GetCurrentDateTime();
+    // Drop TimerTypes that aren't rules or cannot be recorded
+    if (!type->IsTimerRule() && !canRecord)
```

Comment:
For efficiency reasons I would first check IsTimerRule, which is a cheap (local) call and only if it's true also check IsRecordable, which is expensive, because a (potential) remote call.